lib/lighthouse/benefits_intake/service.rb

```diff
@@ -35,15 +35,19 @@ class InvalidDocumentError < StandardError; end
     # Perform the upload to BenefitsIntake
     # parameters should be run through validation functions first, to prevent downstream processing errors
     #
-    # @param metadata [Hash] metadata to be sent with upload
+    # @raise [JSON::ParserError] if metadata is not a valid JSON String
+    # @raise [Errno::ENOENT] if document or each attachment are not valid Files
+    #
+    # @param metadata [JSONString] metadata to be sent with upload, must be valid JSON
     # @param document [String] main document file path
     # @param attachments [Array<String>] attachment file path; optional, default = []
     # @param upload_url [String] override instance upload_url; optional, default = @location
     #
     def perform_upload(metadata:, document:, attachments: [], upload_url: nil)
       upload_url, _uuid = request_upload unless upload_url
 
-      meta_tmp = Common::FileHelpers.generate_temp_file(metadata.to_s, "#{STATSD_KEY_PREFIX}.#{@uuid}.metadata.json")
+      metadata = JSON.parse(metadata)
```

Comment:
Just for my own learning, if JSON.parse fails, will it raise the error here?

Reply:
yes the error will propagate out.  did not seem to make sense to catch the error to then just raise it again